### ISSUE TITLE
When will neuron sdk support GPT-2

### ISSUE THREAD

**Issue by Damiox** [aws/aws-neuron-sdk#118]
This ticket is about this question:
https://github.com/aws/aws-neuron-sdk/issues/88#issuecomment-627453218

**Comment by awsrjh** [comment]
HI Damiox,

We have a set of such models that we are working on or have identified tracked here: 

https://github.com/aws/aws-neuron-sdk/projects/2

Having input from folks like you is also very helpful, so thank you for the heads up. When we have a more concrete answer we will update the roadmap.

thanks

**Comment by misrasaurabh1** [comment]
I tried to compile OpenAI's version of GPT2 with Neuron and after fixing the input feature vector (which you ideally don't want to do), only 5% of operations were placed on Neuron.

**Comment by aws-taylor** [comment]
Hi Saurabh,

Thanks for trying AWS Neuron.  I'm following up on the Tensorflow compilation of this model (the original query was about PyTorch).  On the PyTorch side the Hugging Face model does not currently compile, but we are actively working on resolving the issues impacting GPT2.

If you use the PyTorch coverage tool 

   results = torch.neuron.analyze_model( model, example_inputs )

You can see higher coverage:

WARNING:transformers.modeling_utils:Some weights of GPT2LMHeadModel were not initialized from the model checkpoint at gpt2 and are newly initialized: ['h.0.attn.masked_bias', 'h.1.attn.masked_bias', 'h.2.attn.masked_bias', 'h.3.attn.masked_bias', 'h.4.attn.masked_bias', 'h.5.attn.masked_bias', 'h.6.attn.masked_bias', 'h.7.attn.masked_bias', 'h.8.attn.masked_bias', 'h.9.attn.masked_bias', 'h.10.attn.masked_bias', 'h.11.attn.masked_bias', 'lm_head.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
WARNING:root:The following operations are currently supported in torch-neuron for this model
WARNING:root:aten::permute
WARNING:root:aten::ScalarImplicit
WARNING:root:aten::stack
WARNING:root:aten::slice
WARNING:root:aten::dropout
WARNING:root:aten::add
WARNING:root:aten::unsqueeze
WARNING:root:prim::NumToTensor
WARNING:root:aten::transpose
WARNING:root:aten::t
WARNING:root:aten::softmax
WARNING:root:aten::view
WARNING:root:prim::GetAttr
WARNING:root:aten::matmul
WARNING:root:aten::mul
WARNING:root:prim::ListConstruct
WARNING:root:aten::addmm
WARNING:root:aten::size
WARNING:root:prim::ListUnpack
WARNING:root:aten::sub
WARNING:root:aten::Int
WARNING:root:aten::pow
WARNING:root:prim::TupleConstruct
WARNING:root:aten::arange
WARNING:root:aten::to
WARNING:root:aten::split
WARNING:root:aten::tanh
WARNING:root:aten::contiguous
WARNING:root:prim::TupleUnpack
WARNING:root:prim::Constant
WARNING:root:aten::div
WARNING:root:aten::layer_norm
WARNING:root:The following operations are currently not supported in torch-neuron for this model
WARNING:root:aten::embedding
WARNING:root:aten::floor_divide
WARNING:root:aten::where
WARNING:root:98.65% of operations (3667 of 3717) are supported
{'percent_supported': 98.65482916330375, 'supported_count': 3667, 'total_count': 3717, 'supported_operators': {'aten::permute', 'aten::ScalarImplicit', 'aten::stack', 'aten::slice', 'aten::dropout', 'aten::add', 'aten::unsqueeze', 'prim::NumToTensor', 'aten::transpose', 'aten::t', 'aten::softmax', 'aten::view', 'prim::GetAttr', 'aten::matmul', 'aten::mul', 'prim::ListConstruct', 'aten::addmm', 'aten::size', 'prim::ListUnpack', 'aten::sub', 'aten::Int', 'aten::pow', 'prim::TupleConstruct', 'aten::arange', 'aten::to', 'aten::split', 'aten::tanh', 'aten::contiguous', 'prim::TupleUnpack', 'prim::Constant', 'aten::div', 'aten::layer_norm'}, 'unsupported_operators': ['aten::embedding', 'aten::floor_divide', 'aten::where'], 'operators': ['aten::Int', 'aten::ScalarImplicit', 'aten::add', 'aten::addmm', 'aten::arange', 'aten::contiguous', 'aten::div', 'aten::dropout', 'aten::embedding', 'aten::floor_divide', 'aten::layer_norm', 'aten::matmul', 'aten::mul', 'aten::permute', 'aten::pow', 'aten::size', 'aten::slice', 'aten::softmax', 'aten::split', 'aten::stack', 'aten::sub', 'aten::t', 'aten::tanh', 'aten::to', 'aten::transpose', 'aten::unsqueeze', 'aten::view', 'aten::where', 'prim::Constant', 'prim::GetAttr', 'prim::ListConstruct', 'prim::ListUnpack', 'prim::NumToTensor', 'prim::TupleConstruct', 'prim::TupleUnpack']}

We'll continue to work on compilation on PyTorch and Tensorflow and post updates here.

**Comment by aws-zejdaj** [comment]
Still on the roadmap

**Comment by awsrjh** [comment]
refresh - we have made some progress but are not yet fully ready with support. Please check back or contact us for more assistance.

**Comment by mrnikwaws** [comment]
A quick note that GPT-2 is still work in progress, but has been added to our roadmap [here](https://github.com/aws/aws-neuron-sdk/projects/2)

**Comment by aws-taylor** [comment]
Hello Saurabh,

 We are still working on GPT-2 support. Now that it has been added to our roadmap, I'm going to resolve this issue in favor of https://github.com/aws/aws-neuron-sdk/issues/211. I encourage you to 'Watch' that issue as it moves towards completion.

Regards,
Taylor

**Comment by echatzikyriakidis** [comment]
Is GPT2LMHeadModel (Hugging Face - pytorch) supported currently?

Thanks
